### PR TITLE
ROCm3.8 mivisionx recipe

### PR DESCRIPTION
**Pull request by gkathirv**

### CODE REVIEW

**Comment by gkathirv** [comment]
@adamjstewart @srekolam @arjun-raj-kuppala

File: var/spack/repos/builtin/packages/mivisionx/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    def flag_handler(self, name, flags):
+        spec = self.spec
+        protobuf = spec['protobuf'].prefix.include
+        if name == 'cxxflags':
+            flags.append('-I{0}'.format(protobuf))
+        return (flags, None, None)
+
+    depends_on('ffmpeg@4.1.1', type='build')
+    depends_on('protobuf@3.5.0', type='build')
+    depends_on('opencv@3.4.6~opencl~opencl_svm~openclamdfft~openclamdblas~python', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mivisionx/package.py]: Are these requirements or preferences? `depends_on` only supports hard requirements. If you don't want to build these variants by default, you can override the defaults in `packages.yaml`.

**Review by adamjstewart — COMMENTED** [review]
Is version 1.7 actually needed? Removing it would simplify the package quite a bit.

@haampie may want to review this.

File: var/spack/repos/builtin/packages/mivisionx/package.py (hunk 1 above)
Comment (gkathirv) [reply var/spack/repos/builtin/packages/mivisionx/package.py]: we don't have tagging for ROCm 3.5. The available version during ROCm 3.5 was 1.7. If ROCm 3.5 support is not required for mivisionx, yes we can remove version 1.7.

Comment (gkathirv) [reply var/spack/repos/builtin/packages/mivisionx/package.py]: @adamjstewart yes above dependency is hard requirement for mivisionx package & version 1.7 is needed for ROCm 3.5.

**Comment by srekolam** [comment]
changes look okay to me.

File: var/spack/repos/builtin/packages/mivisionx/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/mivisionx/package.py]: So it won't build at all if someone tries to use OpenCV 3.4.5 or 3.4.7? Or if someone tries to use OpenCV +opencl or +python?

Comment (gkathirv) [reply var/spack/repos/builtin/packages/mivisionx/package.py]: @adamjstewart Able to build with opencv@3.4.6.

**Review by alalazo — APPROVED** [review]

**Pull request merged**